src/Api/PersistentImplementation.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving  #-}
+{-# LANGUAGE ImpredicativeTypes          #-}
+{-# LANGUAGE OverloadedStrings           #-}
+{-# LANGUAGE ScopedTypeVariables         #-}
+{-# LANGUAGE TypeOperators               #-}
+
+{-# OPTIONS_GHC -Wall -Werror -fno-warn-orphans #-}
+
+module Api.PersistentImplementation
```

Comment:
You are right in both cases.

I think we don't use plural in module names. The `Persistent.Implementation.STM` is better.
I agree with you to not to expose the implementation in `src/Persistent.hs`. :)